config/deploy/docker/lib/memory_analyzer.rb

```diff
@@ -61,8 +61,11 @@ def run!
       # recommend 5% above maximum utilization in recent past
       self.recommended_hard_limit_mb = (current_soft_limit_mb * ((_overall_stats.maximum+5.0) / 100.0)).ceil
 
-      # one standard deviation from the mean
-      self.recommended_soft_limit_mb = (current_soft_limit_mb * ((_overall_stats.average + _overall_stats.stddev)/100.0)).ceil
+      # nsd standard deviations from the mean.
+      # 7 out of 63 task definitions had RAM problems when deployed to all
+      # staging installations with 1 stddev. All were delayed jobs.
+      nds = task_definition_name.match?(/dj-(all|long)/) ? 1.25 : 1.0
```

Comment:
@eanders I don't love this solution, but I think the memory analyzer is actually probably working correctly for the most part. The hard limits I don't think were exceeded, and I think it was just the EC2 instances running out of ram (so bigger soft quotas might help). What do you think?

Going to manually adjust the 7 problem task definitions in dynamodb, and we can come back to this in a few weeks (still keeping this only on staging), and after this commit is merged.

Reply:
I think not combining all and long could easily have been the issue.  It also makes some sense that it hit EC2 limits but not hard limits.  That would jive with the discovery I did yesterday.  Especially on smaller instance types, I can see how this would be a bigger issue.